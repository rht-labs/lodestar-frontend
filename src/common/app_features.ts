// All roles defined by the backend to be used in the frontend should be enumerated in this object.

export const APP_FEATURES = {
  admin: 'admin',
  writer: 'writer',
  reader: 'reader',
  notifications: 'notifications',
  engagementCardIcons: 'engagementCardIcons',
  resetUser: 'resetUser',
};

export type AppFeature = keyof typeof APP_FEATURES;
