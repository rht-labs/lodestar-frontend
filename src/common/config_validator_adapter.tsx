import { ValidatorFactory } from '../schemas/validators';
import { EngagementFormConfig } from '../schemas/engagement_config';

export const getValidatorsFromFormOptions = (
  formOptions: EngagementFormConfig = {}
) =>
  Object.keys(formOptions || {}).reduce(
    (previousSchemaGroups, currentGroupKey) => {
      return {
        ...previousSchemaGroups,
        ...Object.keys(formOptions[currentGroupKey] ?? {}).reduce(
          (previousGroupFields, currentFieldKey) => ({
            ...previousGroupFields,
            [currentFieldKey]: (
              formOptions?.[currentGroupKey]?.[currentFieldKey]?.validators ||
              []
            ).map(ValidatorFactory),
          }),
          {}
        ),
      };
    },
    {}
  );
