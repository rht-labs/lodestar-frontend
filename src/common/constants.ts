export const APPLICATION_CONTENT_CONTAINER_ID = 'main-application-content';
