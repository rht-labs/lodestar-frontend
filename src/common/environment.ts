export class Environment {
  public getValue(key: string) {
    return process.env[key];
  }
}
