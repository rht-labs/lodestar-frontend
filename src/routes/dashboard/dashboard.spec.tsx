import React from 'react';
import {Dashboard} from '.';
import {act, render} from '@testing-library/react';
import {TestStateWrapper} from "../../common/test_state_wrapper";
import {MemoryRouter} from 'react-router';

describe('Dashboard Component', () => {
  test('matches snapshot', async () => {
    await act(async () => {
      expect(render(
        <MemoryRouter>
          <TestStateWrapper>
            <Dashboard/>
          </TestStateWrapper>
        </MemoryRouter>)
      ).toMatchSnapshot();
    });
  });
});
