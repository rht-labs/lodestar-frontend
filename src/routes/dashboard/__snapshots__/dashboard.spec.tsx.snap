// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-c-page__main-section pf-m-light"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class="pf-c-title pf-m-2xl"
          >
            Dashboard
          </h1>
          <p
            class=""
            data-pf-content="true"
          >
            This dashboard gives you an overview of all engagements.
          </p>
        </div>
      </section>
      <section
        class="pf-c-page__main-section"
      >
        <div
          class="pf-l-gallery pf-m-gutter"
        >
          <article
            class="pf-c-card pf-m-hoverable pf-m-compact"
            style="min-height: 40vh;"
          >
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-c-empty-state"
              >
                <div
                  class="pf-c-empty-state__content"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-c-empty-state__icon"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 576 512"
                    width="1em"
                  >
                    <path
                      d="M288 32C128.94 32 0 160.94 0 320c0 52.8 14.25 102.26 39.06 144.8 5.61 9.62 16.3 15.2 27.44 15.2h443c11.14 0 21.83-5.58 27.44-15.2C561.75 422.26 576 372.8 576 320c0-159.06-128.94-288-288-288zm0 64c14.71 0 26.58 10.13 30.32 23.65-1.11 2.26-2.64 4.23-3.45 6.67l-9.22 27.67c-5.13 3.49-10.97 6.01-17.64 6.01-17.67 0-32-14.33-32-32S270.33 96 288 96zM96 384c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm48-160c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm246.77-72.41l-61.33 184C343.13 347.33 352 364.54 352 384c0 11.72-3.38 22.55-8.88 32H232.88c-5.5-9.45-8.88-20.28-8.88-32 0-33.94 26.5-61.43 59.9-63.59l61.34-184.01c4.17-12.56 17.73-19.45 30.36-15.17 12.57 4.19 19.35 17.79 15.17 30.36zm14.66 57.2l15.52-46.55c3.47-1.29 7.13-2.23 11.05-2.23 17.67 0 32 14.33 32 32s-14.33 32-32 32c-11.38-.01-20.89-6.28-26.57-15.22zM480 384c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
                      transform=""
                    />
                  </svg>
                  <div
                    class="pf-c-content"
                  >
                    <h1
                      class=""
                      data-cy="numbers_of_l"
                      data-pf-content="true"
                    />
                    <button
                      class="pf-c-button pf-m-link pf-m-inline"
                      data-cy="button_l"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      type="button"
                    >
                      <h2
                        class=""
                        data-pf-content="true"
                      >
                        All Engagements
                      </h2>
                    </button>
                  </div>
                  <div
                    class="pf-c-empty-state__body"
                  >
                    All available engagements in the system. Including upcoming, active and past ones
                  </div>
                </div>
              </div>
            </div>
          </article>
          <article
            class="pf-c-card pf-m-hoverable pf-m-compact"
            style="min-height: 40vh;"
          >
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-c-empty-state"
              >
                <div
                  class="pf-c-empty-state__content"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-c-empty-state__icon"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 64 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M512-73.143c-70.656 0-136.997 13.385-199.022 40.009s-116.151 63.342-162.523 110.007-83.017 101.010-110.007 162.962-40.521 128.366-40.521 199.022c0 70.656 13.531 136.997 40.521 199.022s63.634 116.151 110.007 162.523 100.498 83.017 162.523 110.007 128.366 40.521 199.022 40.521c70.656 0 136.997-13.531 199.022-40.521s116.151-63.707 162.523-110.007c46.299-46.299 83.017-100.498 110.007-162.523s40.521-128.366 40.521-199.022c0-70.656-13.531-136.997-40.521-199.022s-63.707-116.37-110.007-162.962c-46.299-46.665-100.498-83.31-162.523-110.007s-128.366-40.009-199.022-40.009zM361.545 796.379c-46.958-20.334-88.137-48.128-123.465-83.529s-63.195-76.507-83.529-123.465c-20.334-46.958-30.501-97.134-30.501-150.455s10.167-103.643 30.501-150.967c20.334-47.323 48.128-88.503 83.529-123.465s76.507-62.683 123.465-83.017c46.958-20.334 97.134-30.501 150.455-30.501s103.497 10.167 150.455 30.501c46.958 20.334 88.137 47.982 123.465 83.017s63.122 76.142 83.529 123.465c20.334 47.323 30.501 97.646 30.501 150.967s-10.167 103.497-30.501 150.455c-20.334 46.958-48.128 88.137-83.529 123.465s-76.507 63.122-123.465 83.529c-46.958 20.334-97.134 30.501-150.455 30.501s-103.497-10.167-150.455-30.501zM328.997 662.967c0 0-8.338-94.135 73.143-130.633 26.112-11.703 70.875-33.573 72.411-71.68l0.731-1.536c2.267-36.571-19.017-60.928-74.606-80.311-75.557-26.331-71.753-160.037-71.753-160.037v0c-0.073-0.512-0.146-1.024-0.146-1.536v-62.098c0-2.487 0.805-4.535 2.341-6.071s3.584-2.341 6.071-2.341h349.038c2.487 0 4.535 0.805 6.071 2.341s2.341 3.657 2.341 6.071v62.098c0 0.512-0.073 1.024-0.146 1.536v0c0 0 4.096 134.51-73.143 162.597-58.661 21.358-75.41 48.786-74.313 77.751 3.438 30.427 49.225 60.123 74.313 75.118 79.653 47.616 74.167 119.735 73.289 127.927v60.489c0 2.487-0.805 4.535-2.341 6.071s-3.584 2.341-6.071 2.341h-348.965c-2.487 0-4.535-0.805-6.071-2.341s-2.341-3.584-2.341-6.071v-56.174c0-0.878 0.073-2.121 0.293-3.511h-0.146zM527.872 273.188c111.397 0.512 93.696-54.491 93.696-54.491h-219.429c0 55.223 87.625 53.687 87.625 53.687l38.107 0.731z"
                      transform="rotate(180 0 512) scale(-1 1)"
                    />
                  </svg>
                  <div
                    class="pf-c-content"
                  >
                    <h1
                      class=""
                      data-cy="numbers_of_p"
                      data-pf-content="true"
                    />
                    <button
                      class="pf-c-button pf-m-link pf-m-inline"
                      data-cy="button_p"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      type="button"
                    >
                      <h2
                        class=""
                        data-pf-content="true"
                      >
                        Upcoming Engagements
                      </h2>
                    </button>
                  </div>
                  <div
                    class="pf-c-empty-state__body"
                  >
                    Upcoming engagements in the future, and are not launched yet.
                  </div>
                </div>
              </div>
            </div>
          </article>
          <article
            class="pf-c-card pf-m-hoverable pf-m-compact"
            style="min-height: 40vh;"
          >
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-c-empty-state"
              >
                <div
                  class="pf-c-empty-state__content"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-c-empty-state__icon"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 64 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M511.781-73.143c-70.656 0-136.923 13.312-198.875 40.009s-116.078 63.269-162.45 109.934c-46.299 46.665-82.944 100.937-109.934 162.962s-40.448 128.293-40.448 198.875 13.458 136.923 40.448 198.875 63.634 116.078 109.934 162.45c46.299 46.299 100.425 82.944 162.45 109.934s128.293 40.448 198.875 40.448 136.923-13.458 198.875-40.448 116.078-63.634 162.45-109.934c46.299-46.299 82.944-100.425 109.934-162.45s40.448-128.293 40.448-198.875-13.458-136.923-40.448-198.875-63.634-116.297-109.934-162.962-100.425-83.31-162.45-109.934-128.293-40.009-198.875-40.009v0zM361.326 795.94c-46.958-20.334-88.137-48.128-123.465-83.456s-63.122-76.434-83.456-123.465-30.501-97.134-30.501-150.455 10.167-103.643 30.501-150.894c20.334-47.323 48.128-88.43 83.456-123.465s76.434-62.61 123.465-82.944 97.134-30.501 150.455-30.501 103.424 10.167 150.455 30.501c46.958 20.334 88.137 47.982 123.465 82.944s63.122 76.142 83.456 123.465 30.501 97.646 30.501 150.894-10.167 103.424-30.501 150.455c-20.334 46.958-48.128 88.137-83.456 123.465s-76.434 63.122-123.465 83.456-97.134 30.501-150.455 30.501c-53.321 0-103.424-10.167-150.455-30.501v0zM730.697 498.395c1.975-8.119 2.414-19.383-1.17-26.331-7.387-14.263-16.457-26.99-26.185-40.96-83.017-119.296-158.135-200.485-249.71-305.371-17.993-20.626-34.889-13.751-42.057-9.070-15.579 10.24-15.799 29.769-7.314 52.297 9.801 25.966 20.48 57.783 29.842 79.799 22.455 52.736 20.187 58.075 41.911 114.907-52.297-0.1024-127.781-0.951-180.078-1.463-15.067-0.146-33.499 1.902-39.497 16.311-5.778 13.751-0.146 26.039 8.558 38.107 64.146 89.381 172.617 212.626 248.174 294.546 6.51 7.022 30.062 23.333 43.739 21.358 10.825-1.536 21.87 1.243 26.843-9.289 5.851-12.434 2.706-25.746-2.56-37.23-22.016-48.347-45.714-118.638-71.095-174.519 50.615-0.073 144.53-0.073 195.145-0.146 8.558 0 18.432-0.731 23.406-8.046 1.024-1.536 1.755-3.145 2.121-4.901v0z"
                      transform="rotate(180 0 512) scale(-1 1)"
                    />
                  </svg>
                  <div
                    class="pf-c-content"
                  >
                    <h1
                      class=""
                      data-cy="numbers_of_c"
                      data-pf-content="true"
                    />
                    <button
                      class="pf-c-button pf-m-link pf-m-inline"
                      data-cy="button_c"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      type="button"
                    >
                      <h2
                        class=""
                        data-pf-content="true"
                      >
                        Active Engagements
                      </h2>
                    </button>
                  </div>
                  <div
                    class="pf-c-empty-state__body"
                  >
                    Engagements that are already in progress and running at the moment.
                  </div>
                </div>
              </div>
            </div>
          </article>
          <article
            class="pf-c-card pf-m-hoverable pf-m-compact"
            style="min-height: 40vh;"
          >
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-c-empty-state"
              >
                <div
                  class="pf-c-empty-state__content"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-c-empty-state__icon"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 64 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M512-73.143c-70.656 0-136.997 13.385-199.022 40.009s-116.151 63.342-162.523 110.007-83.017 101.010-110.007 162.962-40.521 128.366-40.521 199.022c0 70.656 13.531 136.997 40.521 199.022s63.634 116.151 110.007 162.523 100.498 83.017 162.523 110.007 128.366 40.521 199.022 40.521c70.656 0 136.997-13.531 199.022-40.521s116.151-63.707 162.523-110.007c46.299-46.299 83.017-100.498 110.007-162.523s40.521-128.366 40.521-199.022c0-70.656-13.531-136.997-40.521-199.022s-63.707-116.37-110.007-162.962c-46.299-46.665-100.498-83.31-162.523-110.007s-128.366-40.009-199.022-40.009zM361.545 796.379c-46.958-20.334-88.137-48.128-123.465-83.529s-63.195-76.507-83.529-123.465c-20.334-46.958-30.501-97.134-30.501-150.455s10.167-103.643 30.501-150.967c20.334-47.323 48.128-88.503 83.529-123.465s76.507-62.683 123.465-83.017c46.958-20.334 97.134-30.501 150.455-30.501s103.497 10.167 150.455 30.501c46.958 20.334 88.137 47.982 123.465 83.017s63.122 76.142 83.529 123.465c20.334 47.323 30.501 97.646 30.501 150.967s-10.167 103.497-30.501 150.455c-20.334 46.958-48.128 88.137-83.529 123.465s-76.507 63.122-123.465 83.529c-46.958 20.334-97.134 30.501-150.455 30.501s-103.497-10.167-150.455-30.501zM758.565 330.313c-21.211-45.861-53.248-82.505-96.037-109.934s-89.527-41.179-140.069-41.179c-35.255 0-68.901 6.875-100.937 20.699s-59.758 32.329-83.017 55.589c-23.259 23.259-41.765 50.907-55.589 83.017s-20.699 65.682-20.699 100.937c0 34.597 6.51 67.584 19.456 99.109s30.574 58.807 52.882 81.847 48.859 41.618 79.799 55.735c30.939 14.117 63.707 21.87 98.231 23.259 9.947 0.439 21.358-6.217 25.234-15.506 4.096-9.289 4.681-19.675-2.779-26.697-19.456-17.627-36.571-38.107-46.885-61.513s-17.701-45.787-17.701-71.753c0-33.426 5.998-61.952 22.455-90.185s38.839-50.615 67.072-67.072 59.099-24.722 92.526-24.722c26.624 0 52.443 5.705 77.239 17.262 9.289 4.096 17.408 2.56 24.357-4.389 3.145-3.145 5.12-7.022 5.925-11.483s0.293-8.85-1.536-12.873v0z"
                      transform="rotate(180 0 512) scale(-1 1)"
                    />
                  </svg>
                  <div
                    class="pf-c-content"
                  >
                    <h1
                      class=""
                      data-cy="numbers_of_a"
                      data-pf-content="true"
                    />
                    <button
                      class="pf-c-button pf-m-link pf-m-inline"
                      data-cy="button_a"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      type="button"
                    >
                      <h2
                        class=""
                        data-pf-content="true"
                      >
                        Past Engagements
                      </h2>
                    </button>
                  </div>
                  <div
                    class="pf-c-empty-state__body"
                  >
                    Engagements that are finished, closed or archived.
                  </div>
                </div>
              </div>
            </div>
          </article>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-c-page__main-section pf-m-light"
    >
      <div
        class="pf-c-content"
      >
        <h1
          class="pf-c-title pf-m-2xl"
        >
          Dashboard
        </h1>
        <p
          class=""
          data-pf-content="true"
        >
          This dashboard gives you an overview of all engagements.
        </p>
      </div>
    </section>
    <section
      class="pf-c-page__main-section"
    >
      <div
        class="pf-l-gallery pf-m-gutter"
      >
        <article
          class="pf-c-card pf-m-hoverable pf-m-compact"
          style="min-height: 40vh;"
        >
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-c-empty-state"
            >
              <div
                class="pf-c-empty-state__content"
              >
                <svg
                  aria-hidden="true"
                  class="pf-c-empty-state__icon"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 576 512"
                  width="1em"
                >
                  <path
                    d="M288 32C128.94 32 0 160.94 0 320c0 52.8 14.25 102.26 39.06 144.8 5.61 9.62 16.3 15.2 27.44 15.2h443c11.14 0 21.83-5.58 27.44-15.2C561.75 422.26 576 372.8 576 320c0-159.06-128.94-288-288-288zm0 64c14.71 0 26.58 10.13 30.32 23.65-1.11 2.26-2.64 4.23-3.45 6.67l-9.22 27.67c-5.13 3.49-10.97 6.01-17.64 6.01-17.67 0-32-14.33-32-32S270.33 96 288 96zM96 384c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm48-160c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm246.77-72.41l-61.33 184C343.13 347.33 352 364.54 352 384c0 11.72-3.38 22.55-8.88 32H232.88c-5.5-9.45-8.88-20.28-8.88-32 0-33.94 26.5-61.43 59.9-63.59l61.34-184.01c4.17-12.56 17.73-19.45 30.36-15.17 12.57 4.19 19.35 17.79 15.17 30.36zm14.66 57.2l15.52-46.55c3.47-1.29 7.13-2.23 11.05-2.23 17.67 0 32 14.33 32 32s-14.33 32-32 32c-11.38-.01-20.89-6.28-26.57-15.22zM480 384c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
                    transform=""
                  />
                </svg>
                <div
                  class="pf-c-content"
                >
                  <h1
                    class=""
                    data-cy="numbers_of_l"
                    data-pf-content="true"
                  />
                  <button
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-cy="button_l"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    <h2
                      class=""
                      data-pf-content="true"
                    >
                      All Engagements
                    </h2>
                  </button>
                </div>
                <div
                  class="pf-c-empty-state__body"
                >
                  All available engagements in the system. Including upcoming, active and past ones
                </div>
              </div>
            </div>
          </div>
        </article>
        <article
          class="pf-c-card pf-m-hoverable pf-m-compact"
          style="min-height: 40vh;"
        >
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-c-empty-state"
            >
              <div
                class="pf-c-empty-state__content"
              >
                <svg
                  aria-hidden="true"
                  class="pf-c-empty-state__icon"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 64 1024 1024"
                  width="1em"
                >
                  <path
                    d="M512-73.143c-70.656 0-136.997 13.385-199.022 40.009s-116.151 63.342-162.523 110.007-83.017 101.010-110.007 162.962-40.521 128.366-40.521 199.022c0 70.656 13.531 136.997 40.521 199.022s63.634 116.151 110.007 162.523 100.498 83.017 162.523 110.007 128.366 40.521 199.022 40.521c70.656 0 136.997-13.531 199.022-40.521s116.151-63.707 162.523-110.007c46.299-46.299 83.017-100.498 110.007-162.523s40.521-128.366 40.521-199.022c0-70.656-13.531-136.997-40.521-199.022s-63.707-116.37-110.007-162.962c-46.299-46.665-100.498-83.31-162.523-110.007s-128.366-40.009-199.022-40.009zM361.545 796.379c-46.958-20.334-88.137-48.128-123.465-83.529s-63.195-76.507-83.529-123.465c-20.334-46.958-30.501-97.134-30.501-150.455s10.167-103.643 30.501-150.967c20.334-47.323 48.128-88.503 83.529-123.465s76.507-62.683 123.465-83.017c46.958-20.334 97.134-30.501 150.455-30.501s103.497 10.167 150.455 30.501c46.958 20.334 88.137 47.982 123.465 83.017s63.122 76.142 83.529 123.465c20.334 47.323 30.501 97.646 30.501 150.967s-10.167 103.497-30.501 150.455c-20.334 46.958-48.128 88.137-83.529 123.465s-76.507 63.122-123.465 83.529c-46.958 20.334-97.134 30.501-150.455 30.501s-103.497-10.167-150.455-30.501zM328.997 662.967c0 0-8.338-94.135 73.143-130.633 26.112-11.703 70.875-33.573 72.411-71.68l0.731-1.536c2.267-36.571-19.017-60.928-74.606-80.311-75.557-26.331-71.753-160.037-71.753-160.037v0c-0.073-0.512-0.146-1.024-0.146-1.536v-62.098c0-2.487 0.805-4.535 2.341-6.071s3.584-2.341 6.071-2.341h349.038c2.487 0 4.535 0.805 6.071 2.341s2.341 3.657 2.341 6.071v62.098c0 0.512-0.073 1.024-0.146 1.536v0c0 0 4.096 134.51-73.143 162.597-58.661 21.358-75.41 48.786-74.313 77.751 3.438 30.427 49.225 60.123 74.313 75.118 79.653 47.616 74.167 119.735 73.289 127.927v60.489c0 2.487-0.805 4.535-2.341 6.071s-3.584 2.341-6.071 2.341h-348.965c-2.487 0-4.535-0.805-6.071-2.341s-2.341-3.584-2.341-6.071v-56.174c0-0.878 0.073-2.121 0.293-3.511h-0.146zM527.872 273.188c111.397 0.512 93.696-54.491 93.696-54.491h-219.429c0 55.223 87.625 53.687 87.625 53.687l38.107 0.731z"
                    transform="rotate(180 0 512) scale(-1 1)"
                  />
                </svg>
                <div
                  class="pf-c-content"
                >
                  <h1
                    class=""
                    data-cy="numbers_of_p"
                    data-pf-content="true"
                  />
                  <button
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-cy="button_p"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    <h2
                      class=""
                      data-pf-content="true"
                    >
                      Upcoming Engagements
                    </h2>
                  </button>
                </div>
                <div
                  class="pf-c-empty-state__body"
                >
                  Upcoming engagements in the future, and are not launched yet.
                </div>
              </div>
            </div>
          </div>
        </article>
        <article
          class="pf-c-card pf-m-hoverable pf-m-compact"
          style="min-height: 40vh;"
        >
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-c-empty-state"
            >
              <div
                class="pf-c-empty-state__content"
              >
                <svg
                  aria-hidden="true"
                  class="pf-c-empty-state__icon"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 64 1024 1024"
                  width="1em"
                >
                  <path
                    d="M511.781-73.143c-70.656 0-136.923 13.312-198.875 40.009s-116.078 63.269-162.45 109.934c-46.299 46.665-82.944 100.937-109.934 162.962s-40.448 128.293-40.448 198.875 13.458 136.923 40.448 198.875 63.634 116.078 109.934 162.45c46.299 46.299 100.425 82.944 162.45 109.934s128.293 40.448 198.875 40.448 136.923-13.458 198.875-40.448 116.078-63.634 162.45-109.934c46.299-46.299 82.944-100.425 109.934-162.45s40.448-128.293 40.448-198.875-13.458-136.923-40.448-198.875-63.634-116.297-109.934-162.962-100.425-83.31-162.45-109.934-128.293-40.009-198.875-40.009v0zM361.326 795.94c-46.958-20.334-88.137-48.128-123.465-83.456s-63.122-76.434-83.456-123.465-30.501-97.134-30.501-150.455 10.167-103.643 30.501-150.894c20.334-47.323 48.128-88.43 83.456-123.465s76.434-62.61 123.465-82.944 97.134-30.501 150.455-30.501 103.424 10.167 150.455 30.501c46.958 20.334 88.137 47.982 123.465 82.944s63.122 76.142 83.456 123.465 30.501 97.646 30.501 150.894-10.167 103.424-30.501 150.455c-20.334 46.958-48.128 88.137-83.456 123.465s-76.434 63.122-123.465 83.456-97.134 30.501-150.455 30.501c-53.321 0-103.424-10.167-150.455-30.501v0zM730.697 498.395c1.975-8.119 2.414-19.383-1.17-26.331-7.387-14.263-16.457-26.99-26.185-40.96-83.017-119.296-158.135-200.485-249.71-305.371-17.993-20.626-34.889-13.751-42.057-9.070-15.579 10.24-15.799 29.769-7.314 52.297 9.801 25.966 20.48 57.783 29.842 79.799 22.455 52.736 20.187 58.075 41.911 114.907-52.297-0.1024-127.781-0.951-180.078-1.463-15.067-0.146-33.499 1.902-39.497 16.311-5.778 13.751-0.146 26.039 8.558 38.107 64.146 89.381 172.617 212.626 248.174 294.546 6.51 7.022 30.062 23.333 43.739 21.358 10.825-1.536 21.87 1.243 26.843-9.289 5.851-12.434 2.706-25.746-2.56-37.23-22.016-48.347-45.714-118.638-71.095-174.519 50.615-0.073 144.53-0.073 195.145-0.146 8.558 0 18.432-0.731 23.406-8.046 1.024-1.536 1.755-3.145 2.121-4.901v0z"
                    transform="rotate(180 0 512) scale(-1 1)"
                  />
                </svg>
                <div
                  class="pf-c-content"
                >
                  <h1
                    class=""
                    data-cy="numbers_of_c"
                    data-pf-content="true"
                  />
                  <button
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-cy="button_c"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    <h2
                      class=""
                      data-pf-content="true"
                    >
                      Active Engagements
                    </h2>
                  </button>
                </div>
                <div
                  class="pf-c-empty-state__body"
                >
                  Engagements that are already in progress and running at the moment.
                </div>
              </div>
            </div>
          </div>
        </article>
        <article
          class="pf-c-card pf-m-hoverable pf-m-compact"
          style="min-height: 40vh;"
        >
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-c-empty-state"
            >
              <div
                class="pf-c-empty-state__content"
              >
                <svg
                  aria-hidden="true"
                  class="pf-c-empty-state__icon"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 64 1024 1024"
                  width="1em"
                >
                  <path
                    d="M512-73.143c-70.656 0-136.997 13.385-199.022 40.009s-116.151 63.342-162.523 110.007-83.017 101.010-110.007 162.962-40.521 128.366-40.521 199.022c0 70.656 13.531 136.997 40.521 199.022s63.634 116.151 110.007 162.523 100.498 83.017 162.523 110.007 128.366 40.521 199.022 40.521c70.656 0 136.997-13.531 199.022-40.521s116.151-63.707 162.523-110.007c46.299-46.299 83.017-100.498 110.007-162.523s40.521-128.366 40.521-199.022c0-70.656-13.531-136.997-40.521-199.022s-63.707-116.37-110.007-162.962c-46.299-46.665-100.498-83.31-162.523-110.007s-128.366-40.009-199.022-40.009zM361.545 796.379c-46.958-20.334-88.137-48.128-123.465-83.529s-63.195-76.507-83.529-123.465c-20.334-46.958-30.501-97.134-30.501-150.455s10.167-103.643 30.501-150.967c20.334-47.323 48.128-88.503 83.529-123.465s76.507-62.683 123.465-83.017c46.958-20.334 97.134-30.501 150.455-30.501s103.497 10.167 150.455 30.501c46.958 20.334 88.137 47.982 123.465 83.017s63.122 76.142 83.529 123.465c20.334 47.323 30.501 97.646 30.501 150.967s-10.167 103.497-30.501 150.455c-20.334 46.958-48.128 88.137-83.529 123.465s-76.507 63.122-123.465 83.529c-46.958 20.334-97.134 30.501-150.455 30.501s-103.497-10.167-150.455-30.501zM758.565 330.313c-21.211-45.861-53.248-82.505-96.037-109.934s-89.527-41.179-140.069-41.179c-35.255 0-68.901 6.875-100.937 20.699s-59.758 32.329-83.017 55.589c-23.259 23.259-41.765 50.907-55.589 83.017s-20.699 65.682-20.699 100.937c0 34.597 6.51 67.584 19.456 99.109s30.574 58.807 52.882 81.847 48.859 41.618 79.799 55.735c30.939 14.117 63.707 21.87 98.231 23.259 9.947 0.439 21.358-6.217 25.234-15.506 4.096-9.289 4.681-19.675-2.779-26.697-19.456-17.627-36.571-38.107-46.885-61.513s-17.701-45.787-17.701-71.753c0-33.426 5.998-61.952 22.455-90.185s38.839-50.615 67.072-67.072 59.099-24.722 92.526-24.722c26.624 0 52.443 5.705 77.239 17.262 9.289 4.096 17.408 2.56 24.357-4.389 3.145-3.145 5.12-7.022 5.925-11.483s0.293-8.85-1.536-12.873v0z"
                    transform="rotate(180 0 512) scale(-1 1)"
                  />
                </svg>
                <div
                  class="pf-c-content"
                >
                  <h1
                    class=""
                    data-cy="numbers_of_a"
                    data-pf-content="true"
                  />
                  <button
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-cy="button_a"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    <h2
                      class=""
                      data-pf-content="true"
                    >
                      Past Engagements
                    </h2>
                  </button>
                </div>
                <div
                  class="pf-c-empty-state__body"
                >
                  Engagements that are finished, closed or archived.
                </div>
              </div>
            </div>
          </div>
        </article>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
